doc/en/invocation-fixture.rst

```diff
@@ -0,0 +1,61 @@
+.. _invocation_scoped_fixture:
+
+Invocation-scoped fixtures
+==========================
+
+.. versionadded:: 3.0
+
+.. note::
+    This feature is experimental, so if decided that it brings too much problems
+    or considered too complicated it might be removed in pytest ``3.1``.
+
+    ``tmpdir`` and ``monkeypatch`` might become ``invocation`` scoped
```

Comment:
EDIT: went with initial suggestion and only made `monkeypatch` invocation-scoped, `tmpdir` could be made later on because there are other things to consider.